client/my-sites/plans/main.jsx

```diff
@@ -191,6 +194,26 @@ class Plans extends Component {
 	render() {
 		const { selectedSite, translate, canAccessPlans, currentPlan, eligibleForProPlan } = this.props;
 
+		const maybeRenderLegacyPlanNotice = () => {
```

Comment:
This section is largely duplicated to what we are already doing in here: https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/plans/current-plan/index.jsx#L265

In the name of [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself), let's extract this into a reusable form and plug it back into the both places :)

Reply:
Thanks for the feedback - I agree! :)  I've extracted this to a separate component.